plugin/pom.xml

```diff
@@ -49,6 +49,14 @@
             </artifactItems>
           </configuration>
         </plugin>
+        <plugin>
+          <groupId>org.jenkins-ci.tools</groupId>
+          <artifactId>maven-hpi-plugin</artifactId>
+          <version>3.3</version>
+          <configuration>
+            <warSourceDirectory>../client/target</warSourceDirectory>
+          </configuration>
+        </plugin>
```

Comment:
This change is necessary to allow the Swarm Client JAR to be downloaded from integration test context. Normally, this JAR is made available through the `.hpi` file via the `maven-dependency-plugin` (see lines 25–51 above). However, in integration test context the `.hpi` file is ignored and the test uses the `.hpl` version of the plugin. As such we need to customize the `.hpl` metadata to include the Swarm Client JAR on its classpath.

Reply:
It is not safe, because it also injects the client's classes to the plugin. And not all this libs are guaranteed to be compatible.

Reply:
> It is not safe, because it also injects the client's classes to the plugin. And not all this libs are guaranteed to be compatible.

It is safe, for the reasons explained in [my top-level reply](https://github.com/jenkinsci/swarm-plugin/pull/81#issuecomment-475300696).